lucene/core/src/java/org/apache/lucene/index/FieldInfo.java

```diff
@@ -130,127 +167,255 @@ public boolean checkConsistency() {
+    if (this.indexOptions != IndexOptions.NONE) {
+      verifySamePostingsOptions(
+          this.name,
+          this.storeTermVector,
+          this.omitNorms,
+          this.storePayloads,
+          storeTermVector,
+          omitNorms,
+          storePayloads);
+    }
+    verifySameDocValuesType(this.name, this.docValuesType, docValuesType);
+    verifySameDVGen(this.name, this.dvGen, dvGen);
+    verifySamePointsOptions(
+        this.name,
+        this.pointDimensionCount,
+        this.pointIndexDimensionCount,
+        this.pointNumBytes,
+        dimensionCount,
+        indexDimensionCount,
+        dimensionNumBytes);
+    verifySameVectorOptions(
+        this.name,
+        this.vectorDimension,
+        this.vectorSearchStrategy,
+        vectorDimension,
+        searchStrategy);
+  }
+
+  /**
+   * Very that the provided index options are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySameIndexOptions(String fieldName, IndexOptions io1, IndexOptions io2) {
+    if (io1 != io2) {
+      throw new IllegalArgumentException(
+          "cannot change field \""
+              + fieldName
+              + "\" from index options="
+              + io1
+              + " to inconsistent index options="
+              + io2);
     }
+  }
 
-    if (this.pointDimensionCount == 0 && dimensionCount != 0) {
-      this.pointDimensionCount = dimensionCount;
-      this.pointIndexDimensionCount = indexDimensionCount;
-      this.pointNumBytes = dimensionNumBytes;
-    } else if (dimensionCount != 0
-        && (this.pointDimensionCount != dimensionCount
-            || this.pointIndexDimensionCount != indexDimensionCount
-            || this.pointNumBytes != dimensionNumBytes)) {
+  /**
+   * Very that the provided docValues type are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySameDocValuesType(
+      String fieldName, DocValuesType dv1, DocValuesType dv2) {
+    if (dv1 != dv2) {
       throw new IllegalArgumentException(
           "cannot change field \""
-              + name
+              + fieldName
+              + "\" from doc values type="
+              + dv1
+              + " to inconsistent doc values type="
+              + dv2);
+    }
+  }
+
+  /**
+   * Very that the provided doc values generations are the same
+   *
+   * @throws IllegalArgumentException if they are not the same
+   */
+  public static void verifySameDVGen(String fieldName, long dvg1, long dvg2) {
+    if (dvg1 != dvg2) {
+      throw new IllegalArgumentException(
+          "cannot change field \""
+              + fieldName
+              + "\" from doc values generation="
+              + dvg1
+              + " to inconsistent doc values generation="
+              + dvg2);
+    }
+  }
+
+  /**
+   * Very that the provided posting options are the same
```

Comment:
Several javadocs seem to have this typo.

```suggestion
   * Verify that the provided posting options are the same
```

Reply:
Addressed in 6e7540ebd0ef79536cffabcf0ddc7a592b792252